app/src/main/java/com/babylon/wallet/android/presentation/settings/appsettings/gateways/GatewaysViewModel.kt

```diff
@@ -77,7 +77,7 @@ class GatewaysViewModel @Inject constructor(
             val sanitizedUrl = newUrl.sanitizeAndValidateGatewayUrl(isDevModeEnabled = state.isDeveloperModeEnabled)
             val urlAlreadyAdded = state.gatewayList.any { it.gateway.url == newUrl || it.gateway.url == sanitizedUrl }
             state.copy(
-                newUrlValid = !urlAlreadyAdded && (newUrl.isValidUrl() || sanitizedUrl?.isValidUrl() == true),
+                newUrlValid = !urlAlreadyAdded && (sanitizedUrl != null && sanitizedUrl.isValidUrl()) && newUrl.isValidUrl(),
```

Comment:
Can it not be **sanitizedUrl?.isValidUrl() == true && newUrl.isValidUrl()** ?

Reply:
yep. Updated and added a comment